exec/Chainweb/Gaps.hs

```diff
@@ -52,79 +46,54 @@ gaps env args = do
       logg Info $ fromString $ show e
     Right cutBS -> gapsCut env args cutBS
 
--- TODO Make this configurable?
-blockQueueSize :: Natural
-blockQueueSize = 30
-
 gapsCut :: Env -> GapArgs -> ByteString -> IO ()
 gapsCut env args cutBS = do
-  let curHeight = fromIntegral $ cutMaxHeight cutBS
-      cids = atBlockHeight curHeight $ _env_chainsAtHeight env
-  getBlockGaps env >>= \gapsByChain ->
+  minHeights <- getAndVerifyMinHeights env cutBS
```

Comment:
The min heights query is significantly faster than the gaps query, so moving it to the beginning allows us to fail faster for the user.